src/lexer.mll

```diff
@@ -271,8 +271,82 @@ let raw_markup_target =
 let language_tag_char =
   ['a'-'z' 'A'-'Z' '0'-'9' '_' '-' ]
 
+rule reference_paren_content input start depth_paren depth_curly buffer = parse
+  | '('
+    {
+      Buffer.add_string buffer (Lexing.lexeme lexbuf) ;
+       reference_paren_content input start (depth_paren + 1) depth_curly buffer
+        lexbuf }
+  | '{'
+    {
+      Buffer.add_string buffer (Lexing.lexeme lexbuf) ;
+       reference_paren_content input start depth_paren (depth_curly + 1) buffer
+        lexbuf }
+  | ')'
+    {
+      Buffer.add_string buffer (Lexing.lexeme lexbuf) ;
+      if depth_paren = 0 then ()
+      else
+        ( reference_paren_content input start (depth_paren - 1) depth_curly
+            buffer lexbuf ) }
+  | '}'
+    {
+      Buffer.add_string buffer (Lexing.lexeme lexbuf) ;
+      if depth_curly = 0 then
+        warning
+          input
+          ~start_offset:(Lexing.lexeme_end lexbuf)
+          (Parse_error.not_allowed
+            ~what:"'}' (end of reference)"
+            ~in_what:(
+              Printf.sprintf "'%s' (custom operator)"
+                (Buffer.sub buffer 0 ((Buffer.length buffer) - 1))))
+      else
```

Comment:
This shouldn't be `else` but `;`, continue iterating until parentheses are balanced. (the check above must be `depth_curly <= 0` then)

Reply:
Je ne suis pas sur d'etre d'accord, a priori le check au dessus signifie que la reference est close, mais que l'user a oublié de fermer son operateur. Pourquoi on continuerai dans ce cas ?

Reply:
That's not what's happening. Returning here means stopping counting parentheses but you get back into `reference_content` which will continue reading.

To do what you want, you'd need to implement https://github.com/ocaml-doc/odoc-parser/pull/13#discussion_r977814110